server/src/routes/supplier.ts

```diff
@@ -0,0 +1,72 @@
+import { Request, Response, Router } from "express";
+import {
+  createSupplier,
+  getAllSuppliers,
+  getOneSupplier,
+  updateOneSupplier,
+} from "../controller/supplier";
+import validateData from "../middleware/dataValidation";
+import {
+  supplierUpdateValidator,
+  supplierCreateValidator,
+} from "../validators/supplierValidator";
+
+const router: Router = Router();
+
+router.post(
+  "/",
+  validateData(supplierCreateValidator),
+  async (req: Request, res: Response) => {
+    const data = await createSupplier(req.body);
+
+    if ("errorCode" in data) return res.status(data.errorCode).json(data);
+
+    return res.status(201).json(data);
+  },
+);
+
+router.get("/", async (req: Request, res: Response) => {
+  const data = await getAllSuppliers();
+
+  return res.status(200).json({ data });
+});
+
+router.get("/:uuid", async (req: Request, res: Response) => {
+  const { uuid } = req.params;
+
+  const supplier = await getOneSupplier(uuid);
+
+  if (!supplier)
+    return res.status(404).json({
+      errorKey: "uuid",
+      errorDescription: "Unable to find supplier",
+      errorCode: 404,
+    });
+
+  const data = await getOneSupplier(uuid);
```

Comment:
You are making 2 request to get the same information, you can change the supplier to data and remove this line, and everything will work